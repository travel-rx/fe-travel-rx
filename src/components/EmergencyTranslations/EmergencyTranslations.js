import React, { Component } from 'react';
import { StyleSheet, Text, View, ScrollView, FlatList, TextInput, TouchableOpacity } from 'react-native';
import Footer from '../Footer/Footer';
import { Ionicons } from '@expo/vector-icons'

export default class EmergencyTranslations extends Component {

  static navigationOptions = {
    title: 'Trips',
    headerLeft: null,
    headerStyle: {
      backgroundColor: '#3499AA'
    },
    headerTintColor: '#EBEBEB',
    headerTitleStyle: {
      fontSize: 30,
    },
  };

  render() {
    const { navigation } = this.props;

    return (
      <View style={styles.container}>
        <ScrollView style={styles.tripsContainer}>
          <View style={styles.trip}>
            <Ionicons
              color='#3499AA'
              name='md-globe'
              size={60}
              />
              <View>
                <Text style={styles.tripName}>Meet Itzy!</Text>
                <Text style={styles.tripDestination}>St. Petersbug, FL</Text>
              </View>
          </View>
          <View style={styles.trip}>
            <Ionicons
              color='#3499AA'
              name='md-globe'
              size={60}
              />
              <View>
                <Text style={styles.tripName}>Backpacking Trip</Text>
                <Text style={styles.tripDestination}>Ho Chi Minh, Vietnam</Text>
              </View>
          </View>
        </ScrollView>
        <TouchableOpacity 
          style={styles.button}
          onPress={() => navigation.navigate('AddTrip')}
        >
          <Text style={styles.buttonText}>
            Add Trip
          </Text>
        </TouchableOpacity>
        <Footer navigation={navigation}/>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  heading: {
    color: '#3499AA',
    fontSize: 40,
    paddingTop: 20,

  },
  buttonText: {
    color: '#EBEBEB',
    textAlign: 'center',
    fontSize: 18,
    fontWeight: 'bold',
  },
  tripsContainer: {
    margin: 25,
    marginHorizontal: 15

  },
  trip: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 20
  },
  tripName: {
    color: '#3499AA',
    fontSize: 25,
    fontWeight: 'bold',
    marginLeft: 20
  },
  tripDestination: {
    color: '#212B31',
    fontSize: 15,
    marginLeft: 20
  },
  button: {
    backgroundColor: '#3499AA',
    borderColor: 'white',
    borderWidth: 1,
    borderRadius: 12,
    display: 'flex',
    fontWeight: 'bold',
    height: 45,
    justifyContent: 'center',
    overflow: 'hidden',
    padding: 0,
    width: 250,
    marginBottom: 25
  },
});
