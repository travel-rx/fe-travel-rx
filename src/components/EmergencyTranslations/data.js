const data = {
  spanish: {
    help: 'Ayuda',
    doctor: 'Necesito un médico',
    accident: 'Ha habido un accidente',
    ambulance: 'Necesito una ambulancia',
    medication: 'Medicación',
  },
  french: {
    help: 'Aider',
    doctor: 'j\'ai besoin d\'un médecin',
    accident: 'il y a eu un accident',
    ambulance: 'J\'ai besoin d\'une ambulance',
    medication: 'la médecine',
  }
}

export default data;