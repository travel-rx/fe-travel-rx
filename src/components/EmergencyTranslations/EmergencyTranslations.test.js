import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import { EmergencyTranslations } from './EmergencyTranslations';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('EmergencyTranslations', () => {
  it('should return true', () => {
    expect(true).toEqual(true);
  })
})