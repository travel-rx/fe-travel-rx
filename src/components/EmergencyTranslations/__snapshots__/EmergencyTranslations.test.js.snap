// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmergencyTranslations should match the snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Picker
    mode="dialog"
    onValueChange={[Function]}
    selectedValue="spanish"
    style={
      Object {
        "flex": 0,
        "height": 40.019999999999996,
        "justifyContent": "space-around",
        "marginTop": 106.72,
        "width": 150,
      }
    }
  >
    <PickerItem
      label="Spanish"
      value="spanish"
    />
    <PickerItem
      label="French"
      value="french"
    />
    <PickerItem
      label="German"
      value="german"
    />
    <PickerItem
      label="Chinese"
      value="chinese"
    />
    <PickerItem
      label="Arabic"
      value="arabic"
    />
  </Picker>
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopColor": "#3499AA",
        "borderTopWidth": 2,
        "marginTop": 93.38000000000001,
        "width": 675,
      }
    }
  >
    <View
      style={
        Object {
          "marginBottom": 26.68,
          "marginTop": 40.019999999999996,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3499AA",
            "fontSize": 67,
          }
        }
      >
        Ayuda
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 40,
          }
        }
      >
        "Help"
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 26.68,
          "marginTop": 40.019999999999996,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3499AA",
            "fontSize": 67,
          }
        }
      >
        Necesito un médico
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 40,
          }
        }
      >
        "I need a doctor"
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 26.68,
          "marginTop": 40.019999999999996,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3499AA",
            "fontSize": 67,
          }
        }
      >
        Ha habido un accidente
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 40,
          }
        }
      >
        "There's been an accident"
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 26.68,
          "marginTop": 40.019999999999996,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3499AA",
            "fontSize": 67,
          }
        }
      >
        Necesito una ambulancia
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 40,
          }
        }
      >
        "I need an ambulance"
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 26.68,
          "marginTop": 40.019999999999996,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#3499AA",
            "fontSize": 67,
          }
        }
      >
        Medicación
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 40,
          }
        }
      >
        "Medicine"
      </Text>
    </View>
  </ScrollViewMock>
  <Footer />
</View>
`;
